src/spootnik/reporter/specs.clj

```diff
@@ -35,17 +35,34 @@
                    :spootnik.reporter.config.riemann/defaults
                    :spootnik.reporter.config/tls]))
 
+;; pushgateway
+(s/def :spootnik.reporter.config.pushgateway/job keyword?)
+(s/def :spootnik.reporter.config.pushgateway/name keyword?)
+(s/def :spootnik.reporter.config.pushgateway/type #{:gauge :counter})
+(s/def :spootnik.reporter.config.pushgateway/help string?)
+(s/def :spootnik.reporter.config.pushgateway/label-names (s/coll-of keyword?))
+(s/def :spootnik.reporter.config.pushgateway/metric (s/keys :req-un [:spootnik.reporter.config.pushgateway/name
+                                                                     :spootnik.reporter.config.pushgateway/type
+                                                                     :spootnik.reporter.config.pushgateway/help
+                                                                     :spootnik.reporter.config.pushgateway/label-names]))
+(s/def :spootnik.reporter.config.pushgateway/metrics (s/coll-of :spootnik.reporter.config.pushgateway/metric))
+(s/def :spootnik.reporter.config.pushgateway/pushgateway (s/keys :req-un [:spootnik.reporter.config/host
+                                                                          :spootnik.reporter.config.pushgateway/job]
+                                                                 :opt-un [:spootnik.reporter.config/tls
+                                                                          :spootnik.reporter.config/port]))
+
 ;; single metrics reporter
 (s/def :spootnik.reporter.config.metrics.reporter.config/opts map?)
 (s/def :spootnik.reporter.config.metrics.reporter.config/interval pos-int?)
 (s/def :spootnik.reporter.config.metrics.reporter/config
   (s/keys :req-un []
           :opt-un [:spootnik.reporter.config.metrics.reporter.config/interval
-                   :spootnik.reporter.config.metrics.reporter.config/opts]))
+                   :spootnik.reporter.config.metrics.reporter.config/opts
+                   :spootnik.reporter.config.pushgateway/metrics]))
 
 ;; metrics
 (s/def :spootnik.reporter.config.metrics/reporters
-  (s/map-of #{:graphite :prometheus :riemann :console :jmx}
+  (s/map-of #{:graphite :prometheus :riemann :console :jmx :pushgateway}
             :spootnik.reporter.config.metrics.reporter/config))
```

Comment:
If there is a spec trick that doesn't break existing configurations but lets `:spootnik.reporter.config.pushgateway/metrics` only be used with `:pushgateway`please share your secrets

Reply:
https://clojuredocs.org/clojure.spec.alpha/multi-spec ?

Reply:
I was overthinking this: f8bd391